src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -0,0 +1,178 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+/**
+ * Workflow step to create an ingest pipeline
+ */
+public class CreateIngestPipelineStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateIngestPipelineStep.class);
+    private static final String NAME = "create_ingest_pipeline_step";
+
+    private static final String PIPELINE_ID_FIELD = "id";
+    private static final String DESCRIPTION_FIELD = "description";
+    private static final String MODEL_ID_FIELD = "model_id";
+    private static final String INPUT_FIELD = "input_field_name";
+    private static final String OUTPUT_FIELD = "output_field_name";
+
+    // Client to store a pipeline in the cluster state
+    private final ClusterAdminClient clusterAdminClient;
+    // Client to store response data into global context index
+    private final Client client;
+
+    /**
+     * Instantiates a new CreateIngestPipelineStep
+     *
+     * @param client The client to create a pipeline and store workflow data into the global context index
+     */
+    public CreateIngestPipelineStep(Client client) {
+        this.clusterAdminClient = client.admin().cluster();
+        this.client = client;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+
+        CompletableFuture<WorkflowData> createIngestPipelineFuture = new CompletableFuture<>();
+
+        String pipelineId = null;
+        String description = null;
+        String modelId = null;
+        String inputFieldName = null;
+        String outputFieldName = null;
+        BytesReference configuration = null;
+
+        // Extract required content from workflow data and generate the ingest pipeline configuration
+        for (WorkflowData workflowData : data) {
+
+            Map<String, String> parameters = workflowData.getParams();
+            Map<String, Object> content = workflowData.getContent();
+            logger.debug("Previous step sent params: {}, content: {}", parameters, content);
+
+            if (content.containsKey(PIPELINE_ID_FIELD)) {
+                pipelineId = (String) content.get(PIPELINE_ID_FIELD);
+            }
+            if (content.containsKey(DESCRIPTION_FIELD)) {
+                description = (String) content.get(DESCRIPTION_FIELD);
+            }
+            if (content.containsKey(MODEL_ID_FIELD)) {
+                modelId = (String) content.get(MODEL_ID_FIELD);
+            }
+            if (content.containsKey(INPUT_FIELD)) {
+                inputFieldName = (String) content.get(INPUT_FIELD);
+            }
+            if (content.containsKey(OUTPUT_FIELD)) {
+                outputFieldName = (String) content.get(OUTPUT_FIELD);
+            }
+
+            if (Stream.of(pipelineId, description, modelId, inputFieldName, outputFieldName).allMatch(x -> x != null)) {
```

Comment:
I've never seen this for null checking, I like it!  

However, I don't see an "else" clause here, should we throw an exception if we don't have all the required fields?

Reply:
The exception is thrown below when we check if the configuration is null or not, but ill move this around so its more apparent